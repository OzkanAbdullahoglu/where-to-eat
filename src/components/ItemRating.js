import React from 'react';
import RatingStar from './RatingStar';


const ItemRating = ({ rating, reviewCount }) => {
  const createStars = [...Array(5).keys()];
  let starStyle = '';
  return (
    <span>
      <span className="cards-result-rating" aria-label="4 stars">
        <span className="cards-rating-score">{rating}</span>
        <ol className="cards-rating-stars">
          {createStars.map(
            (item, i) => {
              if (i === Math.floor(rating) && Math.floor(rating) !== rating) {
                starStyle = '-half';
              } else if (i + 1 > Math.floor(rating)) {
                starStyle = '-empty';
              }
              return (
                <RatingStar startStyle={starStyle} key={item} />
              );
            }
          )
          }
        </ol>
      </span>
      <span className="cards-result-num-ratings">{`${reviewCount} reviews`}</span>
    </span>
  );
};
export default ItemRating;
