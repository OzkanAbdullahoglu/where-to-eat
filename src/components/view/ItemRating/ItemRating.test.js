/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ItemRating from './ItemRating';

describe('Item Rating Component', () => {
  const props = {
    rating: 4.5,
    reviewCount: 133,
  };
  const itemRating = shallow(<ItemRating {...props} />);
  it('renders Item Rating Component', () => {
    expect(toJson(itemRating)).toMatchSnapshot();
  });
  it('renders Rating Star Component', () => {
    expect(itemRating.find('RatingStar').exists()).toBe(true);
  });
  it('renders `rating` props', () => {
    const ratingSpan = itemRating.find('.cards-rating-score').text();
    expect(ratingSpan).toEqual((props.rating).toString());
  });
  it('renders `reviewCount` props', () => {
    const reviewCounSpan = itemRating.find('.cards-result-num-ratings').text();
    expect(reviewCounSpan).toEqual(`${props.reviewCount} reviews`);
  });
});
