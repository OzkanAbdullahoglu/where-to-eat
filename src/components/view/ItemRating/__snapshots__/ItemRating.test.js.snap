// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Rating Component renders Item Rating Component 1`] = `
<span>
  <span
    aria-label="4.5 stars"
    className="cards-result-rating"
  >
    <span
      className="cards-rating-score"
    >
      4.5
    </span>
    <ol
      className="cards-rating-stars"
    >
      <RatingStar
        key="0"
        starStyle=""
      />
      <RatingStar
        key="1"
        starStyle=""
      />
      <RatingStar
        key="2"
        starStyle=""
      />
      <RatingStar
        key="3"
        starStyle=""
      />
      <RatingStar
        key="4"
        starStyle="-half"
      />
    </ol>
  </span>
  <span
    className="cards-result-num-ratings"
  >
    133 reviews
  </span>
</span>
`;
